leasing/report/lease/lease_lists_report.py

```diff
@@ -0,0 +1,215 @@
+            label=_("List type"), required=False, choices=LeaseListType.choices()
+        ),
+        "start_date": forms.DateField(label=_("Start date"), required=True),
+        "end_date": forms.DateField(label=_("End date"), required=True),
+    }
+    output_fields = {
+        "lease_id": {"label": _("Lease id"), "source": get_lease_id},
+        "start_date": {"label": _("Start date")},
+        "end_date": {"label": _("End date")},
+    }
+    automatic_excel_column_labels = False
+
+    def get_data(self, input_data):
+        qs = (
+            Lease.objects.filter(
+                (Q(start_date__isnull=True) | Q(start_date__lte=input_data["end_date"]))
+                & (Q(end_date__isnull=True) | Q(end_date__gte=input_data["start_date"]))
+            )
+        ).prefetch_related(
+            "lease_areas",
+            "tenants",
+            "tenants__tenantcontact_set",
+            "tenants__tenantcontact_set__contact",
+            "tenants__rent_shares",
+        )
+
+        return qs
+
+    # 1.1 Vuokraukset, joissa ei ole laskutus käynnissä
+    def get_invoicing_not_enabled(self, report_data):
+        return self.serialize_data(report_data.filter(is_invoicing_enabled=False))
+
+    # 1.2 Vuokraukset, joissa ei ole vuokratiedot kunnossa
+    def get_rent_info_not_complete(self, report_data):
+        return self.serialize_data(report_data.filter(is_rent_info_complete=False))
+
+    # 1.3 Vuokraukset, joissa ei ole vuokratietoja
+    def get_no_rents(self, report_data):
+        return self.serialize_data(
+            report_data.annotate(rent_count=Count("rents")).filter(rent_count=0)
+        )
+
+    # 1.4 Vuokraukset, joissa ei ole eräpäivää (Vuokraukset, joissa vuokralaji on valittu, mutta ei ole eräpäivää)
+    def get_no_due_date(self, report_data):
+        ret = []
+        for lease in report_data.all():
+            due_dates = False
+            for rent in lease.rents.exclude(type__isnull=True):
+                if rent.due_dates.count() > 0:
+                    due_dates = True
+            if not due_dates:
+                ret.append({"lease_id": lease.get_identifier_string()})
+        return ret
+
+    # 1.5 Vuokraukset, joilla on kertakaikkinen vuokra mutta ei ole laskuja.
+    def get_one_time_rents_with_no_invoice(self, report_data):
+        ret = []
+        for lease in report_data.all():
+            for rent in lease.rents.all():
+                if rent.type == RentType.ONE_TIME and lease.invoices.count() == 0:
+                    ret.append({"lease_id": lease.get_identifier_string()})
+        return ret
+
+    # 1.6 Vuokraukset, joissa on virheellinen hallintaosuus
+    def get_incorrect_management_shares(self, report_data, start_date, end_date):
+        ret = []
+        for lease in report_data.all():
+            sum = 0
+            for tenant in lease.tenants.all():
+                sum = sum + tenant.share_numerator / tenant.share_denominator
+            if sum > 1:
```

Comment:
Erroneous shares can also be less than one, so they are only correct when equal to exactly 1.